### DISCUSSION TITLE
Is it better to use the map directive or a normal js array map?

### DISCUSSION THREAD

**Discussion by jdcoldsmith** [lit/lit#4186] [Q&A] [answered]
I noticed that in the Lit documentation under "Templates" -> "Lists" that they use a normal js array map in order to render an unordered list. However, there is also a map directive that accomplishes the same thing. My question is which one is better to use, and why is it better?

Lit documentation use of js map: https://lit.dev/docs/templates/lists/#repeating-templates-with-map
Lit documentation use of map directive: https://lit.dev/docs/templates/directives/#map

**Comment by AndrewJakubowicz** [comment, marked as answer]
The primary difference is that the `map` directive works with any iterable, including arrays, sets, maps, and even generators, whilst `Array.prototype.map` only works on arrays. So it really depends on what iterable you are mapping over.

See https://lit.dev/tutorials/working-with-lists/#2 for more info!

**Reply by jdcoldsmith** [reply]
Ahh that makes perfect sense! Thank you very much!

**Reply by jdcoldsmith** [reply]
My follow up question would be is there any performance difference between using an `Array.prototype.map` and the Lit map directive?

**Reply by augustjk** [reply]
Performance difference would be minimal. The Lit `map` directive literally is just a for...of loop. https://github.com/lit/lit/blob/08d48e3c31da07811b86f0aeae557e140b9c8c08/packages/lit-html/src/directives/map.ts#L23-L33

**Reply by jdcoldsmith** [reply]
Good to know! Thanks @augustjk!
